### ISSUE TITLE
Master node deleted accidentally

### ISSUE THREAD

**Issue by alatui** [kubernetes/kops#3177]
I'd like to know if there is any way to recovery a master node that has been deleted accidentally on AWS.
I still have the other nodes running, but my master was deleted.
I wonder if I could use the state that is store in a S3 bucket to recreate my master again.

**Comment by chrislovecnm** [comment]
Did you delete the ASG or the instance group? 

What is the output of `kops update cluster`?

**Comment by alatui** [comment]
Actually someone deleted the master node(ec2 instance on amazon).
I rebuilt it from scratch anyway.
If something like that happens again is there any procedure to follow?

**Comment by chrislovecnm** [comment]
The asg should recover the node gracefully. If the asg is deleted then perhaps kops update. I have never tried it before ... 

Going to close this, please reopen if you need to

**Comment by gauravarora** [comment]
I had the same problem, the master node was terminated accidentally and the ASG recovered the cluster gracefully.

**Comment by tonybranfort** [comment]
Yep, that worked for me as well (in case anyone runs across this for more info).  My test master somehow got corrupted and all I was getting was 

> unexpected error during validation: error listing nodes: Get https: //api.test.xxxx.net/api/v1/nodes

After some digging realized that the master was somehow corrupt.  I couldn't even ssh into it.  It's a Test region so I tried stopping the master.  When I stopped the master, Kops terminated it and recreated a new one.  And, within a minute or two, kops had the cluster all back up and running nicely.  Not sure why the corrupt master wasn't discovered but I'm happy that kops recovered well without a cluster rebuild.
